### PR TITLE
fix(GC): removing Txs only if they not included to any Chunk

### PR DESCRIPTION
**Pull request by Kouprin**
Fixes https://github.com/nearprotocol/nearcore/issues/2906.
Also adds additional checks to Store Validator (https://github.com/nearprotocol/nearcore/issues/2597).
Also fixes migration bug.

### CODE REVIEW

File: chain/chain/src/store.rs [hunk 1]
```diff
@@ -2133,7 +2148,23 @@ impl<'a> ChainStoreUpdate<'a> {
...
             let key = StatePartKey(sync_hash, shard_id, part_id).try_to_vec()?;
-            self.gc_col(DBCol::ColStateParts, &key);
+            self.gc_col(ColStateParts, &key);
+        }
+        Ok(())
+    }
+
+    pub fn gc_col_transaction(&mut self, tx_hash: CryptoHash) -> Result<(), Error> {
+        let mut refs = self.get_tx_refcount(&tx_hash)?;
+        debug_assert!(refs > 0);
```
Comment (SkidanovAlex) [inline chain/chain/src/store.rs:2158]: If we only want to crash in debug, the below should be `refs.saturating_sub(1)`.

File: chain/chain/src/store_validator/validate.rs [hunk 2]
```diff
@@ -295,6 +295,34 @@ pub(crate) fn chunk_indexed_by_height_created(
...
+
+pub(crate) fn chunk_increase_tx_refcount(
+    sv: &mut StoreValidator,
+    _chunk_hash: &ChunkHash,
+    shard_chunk: &ShardChunk,
+) -> Result<(), StoreValidatorError> {
+    for tx in shard_chunk.transactions.iter() {
+        sv.inner.tx_refcount.entry(tx.get_hash()).and_modify(|x| *x += 1).or_insert(1);
+    }
+    sv.inner.is_tx_refcount_calculated = true;
```
Comment (SkidanovAlex) [inline chain/chain/src/store_validator/validate.rs:322]: What does this variable tracks?
Seems like it gets set to `true`  after just one chunk being processed?

File: chain/chain/src/store.rs (hunk 1 above)
Comment (Kouprin) [reply chain/chain/src/store.rs:2158]: ++

File: chain/chain/src/store_validator/validate.rs (hunk 2 above)
Comment (Kouprin) [reply chain/chain/src/store_validator/validate.rs:337]: Discussed offline.

File: chain/chain/src/store_validator.rs [hunk 3]
```diff
@@ -181,6 +185,15 @@ impl StoreValidator {
                         &shard_chunk,
                         col,
                     );
+                    // Check that all Txs in Chunk are exist
```
Comment (bowenwang1996) [inline chain/chain/src/store_validator.rs:188]:
```suggestion
                    // Check that all Txs in Chunk exist
```

**Pull request merged**